Fix handling of scalar value shard_to_full

sshonTT opened the pull request:
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-xla/issues/1993)

### Problem description
Scalars have rank 0 and can still carry a sharding attribute (replicated). 

### What's changed
This change ensures proper handling of such cases, by early returning without further process to avoid errors such as `Invalid dimension index; got dims: SmallVector([0]), tensor dimension: 0`

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

ssundaramTT commented on ttnn/core/tensor/xtensor/partition.cpp:
```diff
@@ -137,7 +137,22 @@ XtensorAdapter<typename Expression::value_type> concat_ndim(
+        std::vector<DataType> data(expr.begin(), expr.end());
+        std::vector<size_t> shape_vec(expr.shape().cbegin(), expr.shape().cend());
+        return XtensorAdapter<DataType>(std::move(data), std::move(shape_vec));
+    }
```
Seems like the code in this `if` clause is the same as the `if` clause above (`if (num_chunks.empty())`). Might be better to deduplicate by adding this clause to the previous like:

`if (num_chunks.empty() || expressions.size() == 1)`

ssundaramTT reviewed (approved):
Looks good! Not a code owner though, so I don't think my approval does anything lol

sshonTT commented:
Hi @cfjchu, @ayerofieiev-tt — could you take a look at this PR when you have a chance, as code owners?
I’m tagging you since you previously reviewed [this PR](https://github.com/tenstorrent/tt-metal/pull/24317).

sshonTT committed:
Fix handling of scalar sharding

Scalars have rank 0 and can still carry a sharding attribute (typically replicated).
This change ensures proper handling of such cases,  when shard-to-full is called to this tensor.

sshonTT commented:
@philei-tt I had addressed CI issue by rebasing it, so could you review this commit?
